test/e2e/suite-features/consumer.go

```diff
@@ -636,6 +636,208 @@ spec:
 		assert.Contains(ginkgo.GinkgoT(), msg401, "Missing rbac token in request")
 	})
 
+	ginkgo.It("ApisixRoute with hmacAuth consumer", func() {
+		ac := `
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixConsumer
+metadata:
+  name: hmacvalue
+spec:
+  authParameter:
+    hmacAuth:
+      value:
+        access_key: papa
+        secret_key: fatpa
+        algorithm: "hmac-sha256"
+        clock_skew: 0
+`
+		assert.Nil(ginkgo.GinkgoT(), s.CreateResourceFromString(ac), "creating hmacAuth ApisixConsumer")
+
+		// Wait until the ApisixConsumer create event was delivered.
+		time.Sleep(6 * time.Second)
+
+		grs, err := s.ListApisixConsumers()
+		assert.Nil(ginkgo.GinkgoT(), err, "listing consumer")
+		assert.Len(ginkgo.GinkgoT(), grs, 1)
+		assert.Len(ginkgo.GinkgoT(), grs[0].Plugins, 1)
+		hmacAuth, _ := grs[0].Plugins["hmac-auth"].(map[string]interface{})
+		assert.Equal(ginkgo.GinkgoT(), hmacAuth["access_key"], "papa")
+		assert.Equal(ginkgo.GinkgoT(), hmacAuth["secret_key"], "fatpa")
+		assert.Equal(ginkgo.GinkgoT(), hmacAuth["algorithm"], "hmac-sha256")
+		assert.Equal(ginkgo.GinkgoT(), hmacAuth["clock_skew"], float64(0))
+
```

Comment:
Equal(t, expected, actual, msgAndArgs)

You need to modify the order of the parameters